sshd-core/src/main/java/org/apache/sshd/client/channel/PtyCapableChannelSession.java

```diff
@@ -239,6 +255,21 @@ protected void doOpenPty() throws IOException {
             writePacket(buffer);
         }
 
+        if (xForwarding) {
+            if (debugEnabled) {
+                log.debug("doOpenPty({}) Send xForwarding request", this);
+            }
+            Buffer buffer = session.createBuffer(SshConstants.SSH_MSG_CHANNEL_REQUEST);
+            buffer.putInt(getRecipient());
+            buffer.putString("x11-req");
+            buffer.putBoolean(false); // want-reply
+            buffer.putBoolean(false);
+            buffer.putString("MIT-MAGIC-COOKIE-1");
```

Comment:
RFC 4254 does not require MIT-MAGIC-COOKIE-1 to be used. User code will need to be able to configure the authentication protocol name and "cookie" (actually: arbitrary authentication data), for instance after having read it from the .Xauthority file.